### PR TITLE
Fix my courses demos (and a bit of cleanup)

### PR DESCRIPTION
**Pull request by svanherk**
Git wasn't very smart about all the changes here, but this PR:
- Removes the old polyfill
- Removes reference to a deleted demo for the legacy version
- Renames all data files to end in `.json` and point to other `.json` files (needed for the new dev server) - no changes to the actual data

### CODE REVIEW

File: demo/data/completion-tracking/enrollment-completion-tracking-01.json [hunk 1]
```diff
@@ -8,19 +8,19 @@
 			"rel": [
 				"self"
 			],
-			"href": "../data/completion-tracking/enrollment-completion-tracking-04"
```
Comment (annabelsegalld2l) [inline demo/data/completion-tracking/enrollment-completion-tracking-01.json:11]: Something weird happened with this rename... I guess git messed up which files got renamed and changed. But it looks like all the files are there, just in a weird diff order 🤷

File: demo/index.html [hunk 2]
```diff
@@ -6,13 +6,9 @@
 	<body>
 		<h3>Demos</h3>
 		<ul>
-			<h4>My Courses</h4>
 			<li><a href="d2l-my-courses/d2l-my-courses.html">My Courses Demo</a></li>
 			<li><a href="d2l-my-courses/d2l-my-courses-alerts.html">Alerts Demo</a></li>
 			<li><a href="d2l-my-courses/d2l-my-courses-completion-tracking.html">Completion Tracking Demo</a></li>
-
-			<h4>My Courses Legacy</h4>
-			<li><a href="d2l-my-courses-legacy/d2l-my-courses-sort-logic-off.html">Sort Logic Off Demo</a></li>
```
Comment (annabelsegalld2l) [inline demo/index.html:15]: Just double-checking that legacy is for gone for good? 🎉

**Review by annabelsegalld2l — APPROVED** [review]

File: demo/index.html (hunk 2 above)
Comment (svanherk) [reply demo/index.html:15]: Yes! 🎉 I think Noel and I just missed this reference in the big deletion: https://github.com/Brightspace/d2l-my-courses-ui/pull/900

File: demo/data/completion-tracking/enrollment-completion-tracking-01.json (hunk 1 above)
Comment (svanherk) [reply demo/data/completion-tracking/enrollment-completion-tracking-01.json:11]: Ugh yeah that's so weird, _not_ well done GitHub.  Not well done.

**Pull request merged**